Support for 1_16_R2 ?

[01:28:12] [Server thread/INFO]: This server is running CraftBukkit version git-Spigot-379750e-e850144 (MC: 1.16.2) (Implementing API version 1.16.2-R0.1-SNAPSHOT)

```
Caused by: java.lang.ExceptionInInitializerError
	at net.wesjd.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:382) ~[?:?]

Caused by: java.lang.RuntimeException: Your server version isn't supported in AnvilGUI!
	at net.wesjd.anvilgui.version.VersionMatcher.lambda$match$1(VersionMatcher.java:50) ~[?:?]
	at java.util.Optional.orElseThrow(Unknown Source) ~[?:1.8.0_231]
	at net.wesjd.anvilgui.version.VersionMatcher.match(VersionMatcher.java:50) ~[?:?]
	at net.wesjd.anvilgui.AnvilGUI.<clinit>(AnvilGUI.java:35) ~[?:?]
	at net.wesjd.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:382) ~[?:?]
```

I'll try to submit a PR if no-one gets around to it.

--- a/1_16_R2/pom.xml
+++ b/1_16_R2/pom.xml
@@ -1,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>anvilgui-1_16_R2</artifactId>
+
+    <parent>
+        <groupId>net.wesjd</groupId>
+        <artifactId>anvilgui-parent</artifactId>
+        <version>1.3.0</version>
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.spigotmc</groupId>
+            <artifactId>spigot</artifactId>
+            <version>1.16.2-R0.1-SNAPSHOT</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>net.wesjd</groupId>
+            <artifactId>anvilgui-abstraction</artifactId>
+            <version>1.3.0</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
+
+</project>

--- a/1_16_R2/src/main/java/net/wesjd/anvilgui/version/Wrapper1_16_R2.java
+++ b/1_16_R2/src/main/java/net/wesjd/anvilgui/version/Wrapper1_16_R2.java
@@ -1,0 +1,129 @@
+package net.wesjd.anvilgui.version;
+
+import net.minecraft.server.v1_16_R2.*;
+import org.bukkit.craftbukkit.v1_16_R2.CraftWorld;
+import org.bukkit.craftbukkit.v1_16_R2.entity.CraftPlayer;
+import org.bukkit.craftbukkit.v1_16_R2.event.CraftEventFactory;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+
+public class Wrapper1_16_R2 implements VersionWrapper {
+    private int getRealNextContainerId(Player player) {
+        return toNMS(player).nextContainerCounter();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNextContainerId(Player player, Object container) {
+        return ((AnvilContainer) container).getContainerId();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleInventoryCloseEvent(Player player) {
+        CraftEventFactory.handleInventoryCloseEvent(toNMS(player));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendPacketOpenWindow(Player player, int containerId, String guiTitle) {
+        toNMS(player).playerConnection.sendPacket(new PacketPlayOutOpenWindow(containerId, Containers.ANVIL, new ChatMessage(guiTitle)));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendPacketCloseWindow(Player player, int containerId) {
+        toNMS(player).playerConnection.sendPacket(new PacketPlayOutCloseWindow(containerId));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainerDefault(Player player) {
+        toNMS(player).activeContainer = toNMS(player).defaultContainer;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainer(Player player, Object container) {
+        toNMS(player).activeContainer = (Container) container;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainerId(Object container, int containerId) {
+        //noop
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addActiveContainerSlotListener(Object container, Player player) {
+        ((Container) container).addSlotListener(toNMS(player));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Inventory toBukkitInventory(Object container) {
+        return ((Container) container).getBukkitView().getTopInventory();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object newContainerAnvil(Player player, String guiTitle) {
+        return new AnvilContainer(player, guiTitle);
+    }
+
+    /**
+     * Turns a {@link Player} into an NMS one
+     *
+     * @param player The player to be converted
+     * @return the NMS EntityPlayer
+     */
+    private EntityPlayer toNMS(Player player) {
+        return ((CraftPlayer) player).getHandle();
+    }
+
+    /**
+     * Modifications to ContainerAnvil that makes it so you don't have to have xp to use this anvil
+     */
+    private class AnvilContainer extends ContainerAnvil {
+
+        public AnvilContainer(Player player, String guiTitle) {
+            super(getRealNextContainerId(player), ((CraftPlayer) player).getHandle().inventory,
+                    ContainerAccess.at(((CraftWorld) player.getWorld()).getHandle(), new BlockPosition(0, 0, 0)));
+            this.checkReachable = false;
+            setTitle(new ChatMessage(guiTitle));
+        }
+
+        @Override
+        public void e() {
+            super.e();
+            this.levelCost.set(0);
+        }
+
+        public int getContainerId() {
+            return windowId;
+        }
+
+    }
+
+}

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -111,6 +111,11 @@
             <artifactId>anvilgui-1_16_R1</artifactId>
             <version>1.3.0</version>
         </dependency>
+        <dependency>
+            <groupId>net.wesjd</groupId>
+            <artifactId>anvilgui-1_16_R2</artifactId>
+            <version>1.3.0</version>
+        </dependency>
     </dependencies>
 
     <build>

--- a/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
+++ b/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
@@ -34,7 +34,8 @@ public class VersionMatcher {
 			Wrapper1_13_R2.class,
 			Wrapper1_14_R1.class,
 			Wrapper1_15_R1.class,
-			Wrapper1_16_R1.class
+			Wrapper1_16_R1.class,
+			Wrapper1_16_R2.class
 	);
 
 	/**

--- a/pom.xml
+++ b/pom.xml
@@ -31,6 +31,7 @@
         <module>1_14_4_R1</module>
         <module>1_15_R1</module>
         <module>1_16_R1</module>
+        <module>1_16_R2</module>
     </modules>
 
     <repositories>

